edward/util/tensorflow.py

```diff
@@ -310,6 +310,44 @@ def multivariate_rbf(x, y=0.0, sigma=1.0, l=1.0):
       tf.exp(-1.0 / (2.0 * tf.pow(l, 2.0)) * tf.reduce_sum(tf.pow(x - y, 2.0)))
 
 
+def multivariate_rbf_kernel(x, sigma=1.0, l=1.0):
+  """
+  computes the rbf kernel for the whole data x
+  Args:
+  x: the data
+     size: N x D
+  sigma:
+    standard deviation
+  l:
+    scaling parameter
+  Returns:
+     mat: a matrix of size NxN
+  """
+  N = x.get_shape()[0]
+  mat = []
+  for i in range(N):
```

Comment:
is there any way we can convert the multivariate rbf to be vectorized in some way, and not have to loop over every entry in the matrix? i've found this to be the computational bottleneck in our GP experiments.

looking at GPflow's [`kernels.py`](https://github.com/GPflow/GPflow/blob/master/GPflow/kernels.py), i couldn't find out how (or if) they vectorize this operation.